portfolio/src/main/java/com/google/sps/servlets/AuthCheckServlet.java

```diff
@@ -23,7 +23,8 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
     if (!userService.isUserLoggedIn()) {
       JsonObject userInfo = new JsonObject();
       userInfo.addProperty("loggedIn", false);
-      userInfo.addProperty("loginURL", userService.createLoginURL("/"));
+      userInfo.addProperty("loginURL",
+          userService.createLoginURL(request.getHeader("referer")));
```

Comment:
Why this is necessary?

Reply:
So if the user is not logged in, I can create the appropriate login link for that page, which will redirect to the page the user was on, rather than navigating them back to the home screen which could be frustrating.